src/values/struct_value.rs

```diff
@@ -52,6 +52,17 @@ impl<'ctx> StructValue<'ctx> {
         unsafe { Some(BasicValueEnum::new(LLVMGetOperand(self.as_value_ref(), index))) }
     }
 
+    /// Sets the value of a field belonging to this `StructValue`.
+    pub fn set_field_at_index<BV: BasicValue<'ctx>>(self, index: u32, val: BV) -> bool {
+        if index >= self.count_fields() {
+            return false;
+        }
+
+        unsafe { LLVMSetOperand(self.as_value_ref(), index, val.as_value_ref()) }
```

Comment:
What happens if the field and value are of different types? Is that UB? Or just a IR type error?

Reply:
Ah, yep it segfaults. I added type checking.